app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -16,6 +16,8 @@ def index
                             client_id: @authorize_form.client_id,
                             errors: @authorize_form.errors.messages)
 
+      return create if already_allowed?
```

Comment:
It's not just redirecting to the URI in the registered redirect_uri, it's adding in params to that redirect URL from the form and then performing the side effects (like making sure the `code` param in the redirect URI corresponds to a valid session ID, etc etc)